### ISSUE TITLE
_

### ISSUE THREAD

**Issue by briancullinan** [WebAssembly/design#1455]
_

**Comment by juj** [comment]
> Forkers like this one https://github.com/juj/emscripten-scummvm [ ... ]. The error is in the project founder's ability to upstream their discoveries.

Nope. If you browse open source projects and judge work you would like to see done in them be a failure of other people's abilities, your thinking is where the error is.

**Comment by briancullinan** [comment]
> > Forkers like this one https://github.com/juj/emscripten-scummvm [ ... ]. The error is in the project founder's ability to upstream their discoveries.
> 
> Nope. If you browse open source projects and judge work you would like to see done in them be a failure of other people's abilities, your thinking is where the error is.

Obviously this has been merged in upstream, is that what you are trying to shame me with?

I've literally stated that I'm trying to help others and you refuse to acknowledge that.
"I am going through a bunch of engine ports from years ago, updating the toolset to use the new clang-LLVM system."

Sorry, I was too late to help you.

**Comment by juj** [comment]
> I see porting open-source apps to web as a way to honor other people's hard work. But for some reason @juj thinks I'm judging people instead of asking for more ideas on how to contribute.

Thank you. Now I understand, and such effort is appreciated. Your original post instead read like it was judging off people, thanks for the edit.

> Obviously this has been merged in upstream

Emscripten-scummvm has not been merged to upstream scummvm to the best of my knowledge. Nor should it be. Best approach would be to redo the port from scratch in a completely new way, using the newer Emscripten PROXY_TO_PTHREAD build mode that did not exist when Emscripten-scummvm was worked on a decade ago.

**Comment by briancullinan** [comment]
My dad turned down a software contract for a 3D manufacturing-ERP system. I take it as a deeply personal failure of mine to write compelling software.

I wonder if I might even pay for Google advertising to improve this list:

![Screen Shot 2022-08-14 at 2 38 47 PM](https://user-images.githubusercontent.com/25057134/184555899-2cbc96ed-7b6a-438f-b07e-e43f8c5b14d5.png)

As a part of a campaign, anyone could come up with a dozen search terms I can add to the list. I don't even know what to do anymore, I don't have to money to invest in my peer's good ideas, only the time.
GitHub Actions has helped me validate my effort, and somehow I have free hosting, it is wonderful. I've been priced out of Amazon and Google already.

**Comment by briancullinan** [comment]
> 
> Thank you. Now I understand, and such effort is appreciated. Your original post instead read like it was judging off people, thanks for the edit.
> 

I'm sorry. I wasn't about to call it a "first attempt" since it's being worked on. I only meant to entertain the idea that "project layout" affects other people's perception of complexity.

> 
> Has not been merged to upstream scummvm to the best of my knowledge. Nor should it be.
> 

I guess. I've seen other projects with /win/ and /nix/ system folders for platform code, I made a /wasm/ folder for my .js platform code. But the project owner said to keep it in a separate repo to support your idea. ioq3 on the other hand was totally open to merging the WASI specific stuff with the codebase.

In my own first attempt I tried to reduce the number of code changes based on inolen's "first->nth attempt" on QuakeJS and update his work to the latest release. I guess all I'm trying to point out is that there are at least 2 different ways of thinking about WebAssembly porting, and maybe if `/escripten-ports/` had started out by merging upstream it would have set the precedent. I'm not saying they didn't anything inherently wrong, I'm just trying to understand how people are trying to reduce the effects of "web assembly is still a bit annoying to work with".

I don't know what is more nuanced, writing code or English. Multiple people have come to me asking for their mods and games to be ported to web, I know there must be something here. I DID NOT expect GitHub to email you and make it look like I'm shaming your work.
